modules/rostests/apitests/ntdll/NtNotifyChangeMultipleKeys.c

```diff
@@ -0,0 +1,397 @@
+    TEST_ERROR(state, Status, STATUS_PENDING);
+
+    /* Check event state */
+    Status = WaitForSingleObject(state->EventHandle, 0);
+    TEST_ERROR(state, Status, WAIT_TIMEOUT);
+
+    /* Make change to the registry key */
+    Status = NtSetValueKey(state->KeyHandle, &ValueName, 0, REG_DWORD, &ValueData1, sizeof(ValueData1));
+    TEST_ERROR(state, Status, STATUS_SUCCESS);
+    SleepEx(100, TRUE);
+    Status = NtSetValueKey(state->KeyHandle, &ValueName, 0, REG_DWORD, &ValueData2, sizeof(ValueData2));
+    TEST_ERROR(state, Status, STATUS_SUCCESS);
+
+    /* Verify that the event is signaled */
+    Status = WaitForSingleObjectEx(state->EventHandle, 100, TRUE);
+    TEST_ERROR(state, Status, WAIT_OBJECT_0);
+
+    FINALIZE_TEST(state);
+}
+
+START_SUBTEST(WatchSubtree)
+{
+    NTSTATUS Status;
+
+    UNICODE_STRING ValueName;
+    RtlInitUnicodeString(&ValueName, L"SubtreeTestValue");
+
+    DWORD ValueData1 = 0x12345678;
+    DWORD ValueData2 = 0x87654321;
+
+    INIT_TEST(state, Status);
+
+    /* Create event */
+    state->EventHandle = CreateEvent(NULL, FALSE, FALSE, NULL);
+    ok(state->EventHandle != NULL, "Failed to create event\n");
+
+    /* Start watching for changes */
+    Status = NtNotifyChangeMultipleKeys(state->KeyHandle, 0, NULL, state->EventHandle, NULL, NULL, &state->IoStatusBlock, REG_NOTIFY_CHANGE_LAST_SET, TRUE, NULL, 0, TRUE);
+    TEST_ERROR(state, Status, STATUS_PENDING);
+
+    /* Check event state */
+    Status = WaitForSingleObject(state->EventHandle, 0);
+    TEST_ERROR(state, Status, WAIT_TIMEOUT);
+
+    /* Make change to the subkey */
+    Status = NtSetValueKey(state->SubKeyHandle, &ValueName, 0, REG_DWORD, &ValueData1, sizeof(ValueData1));
+    TEST_ERROR(state, Status, STATUS_SUCCESS);
+    SleepEx(100, TRUE);
+    Status = NtSetValueKey(state->SubKeyHandle, &ValueName, 0, REG_DWORD, &ValueData2, sizeof(ValueData2));
+    TEST_ERROR(state, Status, STATUS_SUCCESS);
+
+    /* Verify that the event is signaled */
+    Status = WaitForSingleObjectEx(state->EventHandle, 100, TRUE);
+    TEST_ERROR(state, Status, WAIT_OBJECT_0);
+
+    FINALIZE_TEST(state);
+}
+
+START_SUBTEST(WatchSecondaryKey)
+{
+    NTSTATUS Status;
+
+    UNICODE_STRING ValueName;
+    RtlInitUnicodeString(&ValueName, L"SecondaryKeyTestValue");
+
+    DWORD ValueData1 = 0x12345678;
+    DWORD ValueData2 = 0x87654321;
+
+    HANDLE SecondaryKey;
+
+    INIT_TEST(state, Status);
+
+    /* Create event */
+    state->EventHandle = CreateEvent(NULL, FALSE, FALSE, NULL);
+    ok(state->EventHandle != NULL, "Failed to create event\n");
+
+    /* Start watching for changes */
+    OBJECT_ATTRIBUTES SubordinateObjects[1];
+    InitializeObjectAttributes(&SubordinateObjects[0], &state->SecondaryKeyName, OBJ_CASE_INSENSITIVE, NULL, NULL);
+    Status = NtNotifyChangeMultipleKeys(state->KeyHandle,
+                                        sizeof(SubordinateObjects)/sizeof(OBJECT_ATTRIBUTES),
```

Comment:
Use `_countof`, `ARRAYSIZE` or whatever

Reply:
Yup, thanks for the suggestion 
I've forgotten that ARRAYSIZE exists too